File: rust/name/name.rs
use std::env;
use std::io;

fn main() {
    let args: Vec<String> = env::args().collect();

    if args.len() > 1 {
        println!("Hi, {}!", args[1]);
    } else {
        println!("Please enter your name: ");
        let mut name = String::new();
        io::stdin().read_line(&mut name).expect("Failed to read line");

        let name = name.trim();

        println!("Hi, {}", name);
    }
}

File: go/name/name.go
package main

import (
	"fmt"
	"bufio"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	fmt.Print("Enter your name: ")
	name, err := reader.ReadString('\n')
	if err != nil {
		fmt.Println("Error reading input: ", err)
		return
	}

	fmt.Println("Hi", name)
	
	
}
